src/NUnitEngine/nunit.engine.api/TestPackage.cs

```diff
@@ -63,8 +61,19 @@ public TestPackage(string filePath)
         /// <param name="testFiles"></param>
         public TestPackage(IList<string> testFiles)
         {
+            ID = GetNextID();
+            SubPackages = new List<TestPackage>();
+            Settings = new Dictionary<string, object>();
+
             foreach (string testFile in testFiles)
-                Add(Path.GetFullPath(testFile));
+                SubPackages.Add(new TestPackage(testFile));
+        }
+
+        private static int _nextID = 0;
+
+        private string GetNextID()
+        {
+            return unchecked(_nextID++).ToString();
```

Comment:
Actually, it's carried over from some very old code that started out in .NET 1.0. It's been copied forward for years and I duplicated it here.  It was used for numbering tests and the idea is that we don't want an excess of  tests to cause the whole run to fail, even if it results in duplicate ids. IIRC, 'unchecked' was needed for some compilers but not others.

In reality, it's not needed here. We might have more than MAXINT tests but we aren't going to have that many packages in a run. Probably I should remove it since it's potentially confusing.